packages/common/refresh-runtime.js

```diff
@@ -565,6 +580,13 @@ export function registerExportsForReactRefresh(filename, moduleExports) {
       // The register function has an identity check to not register twice the same component,
       // so this is safe to not used the same key here.
       register(exportValue, filename + ' export ' + key)
+    } else if (isCompoundComponent(exportValue)) {
+      for (const subKey in exportValue) {
+        register(
+          exportValue[subKey],
+          filename + ' export ' + key + '$' + subKey,
```

Comment:
I think this will conflict when
```js
export const foo = { bar: component1 }
export const foo$bar = component2
```
. I think this can be fixed by picking a character that cannot be used for identifiers like `-` or ` ` instead of `$`.